packages/components/styled-system/src/define-styles.ts

```diff
@@ -12,11 +12,13 @@ export type StyleFunctionProps = {
   [key: string]: any
 }
 
-export type SystemStyleFunction = (
-  props: StyleFunctionProps,
+export type SystemStyleFunction<T extends StyleFunctionProps> = (
```

Comment:
Since we know that SystemStyleFunction can only accept T is superset of `StyleFunctionProps` otherwise it will default to `StyleFunctionProps`
Hence we can go for  - 

export type SystemStyleFunction<T extends StyleFunctionProps = StyleFunctionProps>...

Reply:
Makes sense. Will do this change.